pilot/pkg/bootstrap/server.go

```diff
@@ -393,6 +396,21 @@ func (s *Server) Start(stop <-chan struct{}) error {
 		}
 	}()
 
+	if s.HTTP2Listener != nil {
```

Comment:
do we want a separate listener? I think we can do a single listener that supports h1 and h2 ?

Reply:
It is a virtual/fake listener - we do the same thing for gRPC. 

In reality there is a single listener on 8080 ( the default knative port), and based on detection the HTTPListenr, HTTP2Listener or GRPCListener get called.

Reply:
Oh, right. That makes sense.

Do we even need cmux to do it though? or just replace  the existing http.Server with

```
	h2s := &http2.Server{}
	s.server = &http.Server{
		Handler: h2c.NewHandler(&httpHandler{
			Config: s.Config,
		}, h2s),
	}
```

Reply:
Looking at `h2c.NewHandler` its pretty much doing the same as cmux, so maybe it doesn't matter

Reply:
We need cmux for gRPC. We could also use the gRPC handler - don't remember why we switched back to using the gRPC h2 stack - but that's the main reason for cmux ( right now ).